aws/data_source_aws_sfn_activity.go

```diff
@@ -0,0 +1,74 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/sfn"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceAwsSfnActivity() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsSfnActivityRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
```

Comment:
We will want to ensure that at least one of these configurations is supplied and likely not both. This can be done via the `ExactlyOneOf` schema configuration (these attributes should also be marked `Computed`):

```go
			"name": {
				Type:     schema.TypeString,
				Optional: true,
				Computed: true,
				ExactlyOneOf: []string{
					"arn",
					"name",
				},
			},
			"arn": {
				Type:     schema.TypeString,
				Optional: true,
				Computed: true,
				ExactlyOneOf: []string{
					"arn",
					"name",
				},
			},
```